ci/env-msys.inc.sh

```diff
@@ -6,8 +6,8 @@
 export CFLAGS="${CFLAGS}"
 export CXXFLAGS="${CXXFLAGS}"
 export LDFLAGS="${LDFLAGS}"
-export CC=gcc
-export CXX=g++
+export CC=${CC-gcc} 
+export CXX=${CXX-g++}
```

Comment:
If `$CC` (and `$CXX`) are set to empty strings, after `CC=${CC-gcc}`, `$CC` would still remain as empty strings.  Is this intended?  (similar with `$CXX`)

Reply:
Also, there is a trailing white space on line 9.

Reply:
This is protection against CC/CXX unset.    
CC=""  will force cmake default which would be gcc.   If CC is unset it will also force cmake default but it may fail at this line depending on shell settings.   

Actually it is an interesting issue related to CI logic consistency. 
CI logic is distributed accross _YAML-shell scripts-cmake_    If thess shell scripts are always called from GHA you can ensure that CC and other variablle are set in YAML.   Would you like to rely on this scenario only or hacve some extra logic that make scripts work reliably without GHA ?

Reply:
I see, I didn't know about cmake's behaviour on empty string CC :)  Thanks!

Before your reply, I would have preferred `export CC=${CC:-gcc}` with the extra colon, so `CC` would actually explicitly take on the value of `"gcc"` instead of becoming an empty string when `CC` is originally unset.

Observe:

```console
$ unset TEST
$ echo "${TEST-this is a test}"
this is a test
$ TEST=
$ echo "${TEST-this is a test}"

$ echo  "${TEST:-this is a test}"
this is a test
```

> Actually it is an interesting issue related to CI logic consistency.
CI logic is distributed accross YAML-shell scripts-cmake If thess shell scripts are always called from GHA you can ensure that CC and other variablle are set in YAML. Would you like to rely on this scenario only or hacve some extra logic that make scripts work reliably without GHA ?

For now, I think we can safely assume that these scripts are always run within GHA, as long as they live inside `ci/`.

Reply:
Actually I also put _export CC=${CC:-gcc}_ first but then I relized that if empty string comes from the upstream it does not hurt but may have some idea behind it.